pkg/antctl/raw/traceflow/command.go

```diff
@@ -0,0 +1,397 @@
+		fmt.Println("Please provide source and destination.")
+		return nil
+	}
+
+	kubeconfigPath, err := cmd.Flags().GetString("kubeconfig")
+	if err != nil {
+		return err
+	}
+	kubeconfig, err := runtime.ResolveKubeconfig(kubeconfigPath)
+	if err != nil {
+		return err
+	}
+	setupKubeconfig(kubeconfig, &v1alpha1.SchemeGroupVersion)
+
+	k8sclient, err := kubernetes.NewForConfig(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("error when creating kubernetes clientset: %w", err)
+	}
+	client, err := clientset.NewForConfig(kubeconfig)
+	if err != nil {
+		return fmt.Errorf("error when creating clientset: %w", err)
+	}
+
+	tf, err := newTraceflow(k8sclient)
+	if err != nil {
+		return fmt.Errorf("error when filling up traceflow config: %w", err)
+	}
+
+	ctx, _ := context.WithTimeout(context.Background(), 5*time.Second)
+	if _, err = client.OpsV1alpha1().Traceflows().Create(ctx, tf, metav1.CreateOptions{}); err != nil {
+		return fmt.Errorf("error when creating traceflow, is traceflow feature gate enabled? %w", err)
+	}
+	defer func() {
+		if option.waiting {
+			if err = client.OpsV1alpha1().Traceflows().Delete(context.TODO(), tf.Name, metav1.DeleteOptions{}); err != nil {
+				klog.Errorf("error when deleting traceflow: %+v", err)
+			}
+		}
+	}()
+
+	if !option.waiting {
+		return nil
+	}
+
+	if err := wait.Poll(1*time.Second, 15*time.Second, func() (bool, error) {
+		tf, err := client.OpsV1alpha1().Traceflows().Get(context.TODO(), tf.Name, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if tf.Status.Phase != v1alpha1.Succeeded {
+			return false, nil
+		}
+		if err := output(tf); err != nil {
+			return false, fmt.Errorf("error when outputing result: %w", err)
+		}
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("error when retrieving traceflow: %w", err)
+	}
+
+	return nil
+}
+
+func setupKubeconfig(kubeconfig *rest.Config, groupVersion *schema.GroupVersion) {
+	kubeconfig.APIPath = "/apis"
+	kubeconfig.GroupVersion = groupVersion
+	kubeconfig.NegotiatedSerializer = scheme.Codecs.WithoutConversion()
+	kubeconfig.Insecure = true
+	kubeconfig.CAFile = ""
+	kubeconfig.CAData = nil
+}
+
+func newTraceflow(client kubernetes.Interface) (*v1alpha1.Traceflow, error) {
+	var name string
+	var src v1alpha1.Source
+	split := strings.Split(option.source, "/")
+	if len(split) == 1 {
+		src.Namespace = "default"
+		src.Pod = split[0]
+	} else if len(split) == 2 && len(split[0]) != 0 && len(split[1]) != 0 {
+		src.Namespace = split[0]
+		src.Pod = split[1]
+	} else {
+		return nil, fmt.Errorf("source should be in the format of namespace/pod or pod")
+	}
+
+	var dst v1alpha1.Destination
+	dstIP := net.ParseIP(option.destination)
+	if dstIP != nil {
+		dst.IP = dstIP.String()
+		name = getTFName(fmt.Sprintf("%s-%s-to-%s", src.Namespace, src.Pod, dst.IP))
+	} else {
+		var isPod bool
+		var err error
+		split = strings.Split(option.destination, "/")
+		if len(split) == 1 {
```

Comment:
I think the following `if` blocks can be simplified since the only different is the namespace.
Consider following workflow:
```
if len == 1 {
    setNameSpace
} else if len == 2 {
    setNameSapce
} else {
   error
}
original codes in `if` blocks. Also, we can just check `isPod` one time by using this flow.
```

Reply:
Good idea! Updated.